sql/parser2/eval.go

```diff
@@ -639,3 +687,23 @@ func evalCaseExpr(expr *CaseExpr, env Env) (Datum, error) {
 	}
 	return null, nil
 }
+
+func evalTupleEQ(ldatum, rdatum Datum) (Datum, error) {
+	left := ldatum.(dtuple)
+	right := rdatum.(dtuple)
+	if len(left) != len(right) {
+		return dbool(false), nil
+	}
+	for i, lval := range left {
```

Comment:
Stylistically a normal for-loop looks better. My understanding is that range loops which don't index into the slice being ranged avoid slice bounds checks. It definitely doesn't matter in this code. I'll change it.